services/indexes/pvm/writer.go

```diff
@@ -461,102 +489,104 @@ func (w *Writer) indexTransaction(ctx services.ConsumerCtx, blkID ids.ID, tx pla
 			if err != nil {
 				return err
 			}
 		}
-	case *platformvm.UnsignedAddSubnetValidatorTx:
+	case *txs.AddSubnetValidatorTx:
 		baseTx = castTx.BaseTx.BaseTx
 		typ = models.TransactionTypeAddSubnetValidator
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-	case *platformvm.UnsignedAddDelegatorTx:
+	case *txs.AddDelegatorTx:
 		baseTx = castTx.BaseTx.BaseTx
 		outs = &avaxIndexer.AddOutsContainer{
-			Outs:    castTx.Stake,
+			Outs:    castTx.StakeOuts,
 			Stake:   true,
 			ChainID: w.chainID,
 		}
 		typ = models.TransactionTypeAddDelegator
-		err = w.InsertTransactionValidator(ctx, baseTx.ID(), castTx.Validator)
+		err = w.InsertTransactionValidator(ctx, tx.ID(), castTx.Validator)
 		if err != nil {
 			return err
 		}
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-		if castTx.RewardsOwner != nil {
-			err = w.insertTransactionsRewardsOwners(ctx, castTx.RewardsOwner, baseTx, castTx.Stake)
-			if err != nil {
-				return err
-			}
+		err = w.insertTransactionsRewardsOwners(ctx, tx.ID(), castTx.DelegationRewardsOwner, baseTx, castTx.StakeOuts)
+		if err != nil {
+			return err
 		}
-	case *platformvm.UnsignedCreateSubnetTx:
+	case *txs.CreateSubnetTx:
 		baseTx = castTx.BaseTx.BaseTx
 		typ = models.TransactionTypeCreateSubnet
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-	case *platformvm.UnsignedCreateChainTx:
+	case *txs.CreateChainTx:
 		baseTx = castTx.BaseTx.BaseTx
 		typ = models.TransactionTypeCreateChain
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-	case *platformvm.UnsignedImportTx:
+	case *txs.ImportTx:
 		baseTx = castTx.BaseTx.BaseTx
 		ins = &avaxIndexer.AddInsContainer{
 			Ins:     castTx.ImportedInputs,
 			ChainID: castTx.SourceChain.String(),
 		}
 		typ = models.TransactionTypePVMImport
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-	case *platformvm.UnsignedExportTx:
+	case *txs.ExportTx:
 		baseTx = castTx.BaseTx.BaseTx
 		outs = &avaxIndexer.AddOutsContainer{
 			Outs:    castTx.ExportedOutputs,
 			ChainID: castTx.DestinationChain.String(),
 		}
 		typ = models.TransactionTypePVMExport
-		err = w.InsertTransactionBlock(ctx, baseTx.ID(), blkID)
+		err = w.InsertTransactionBlock(ctx, tx.ID(), blkID)
 		if err != nil {
 			return err
 		}
-	case *platformvm.UnsignedAdvanceTimeTx:
+	case *txs.AdvanceTimeTx:
+	case *txs.RemoveSubnetValidatorTx:
+	case *txs.TransformSubnetTx:
+	case *txs.AddPermissionlessValidatorTx:
+	case *txs.AddPermissionlessDelegatorTx:
 		return nil
```

Comment:
I don't think this is doing what you wanted, this only returns `nil` for `*txs.AddPermissionlessDelegatorTx`. The rest will exit the `switch` and then be `InsertTransaction`ed

Reply:
updated